### DISCUSSION TITLE
一些建议

### DISCUSSION THREAD

**Discussion by Bian-Sh** [unitycoder/UnityLauncherPro#177] [Ideas]
一些建议：
1.标记指定工程为：喜欢，以便收藏这个工程。
2.根据用户开启工程频度来判定星级。星级范围：0~5.
3.创建支持折叠和展开的分组。
4.工程列表子项支持拖曳，便于指定分组。

some advises:
1. Mark the designated project as: Like, in order to collect this project.
2. Judging the star rating based on the frequency with which the user opens the project. Star range: 0 ~ 5.
3. Create groups that support folding and unfolding.
4. The project list sub-item supports drag and drop, which is convenient for specifying grouping.

The functional suggestions I gave seem to be found in any music player.

**Comment by unitycoder** [comment]
some comments:
i like the designated or favorite project idea, to quickly work on your main projects,
not sure how to show that in UI though.. maybe favourited project is always on top?

currently the list is taken directly from unity recent projects registry,
so 2-4 would require saving that sorting and grouping data somewhere else.. (and keeping it synced with unity registry?)

recent list fetch is at https://github.com/unitycoder/UnityLauncherPro/blob/master/UnityLauncherPro/GetProjects.cs

thanks!

**Comment by Bian-Sh** [comment]
![image](https://user-images.githubusercontent.com/24520547/79069346-0d031800-7d00-11ea-8323-9b77f3ea274f.png)
1. the pic above maybe a way you can show "Favorite " Folder and the "I like" mark in UI,
2. I've already read you code and knew how you catch those projects , I think there is no need to keep it synced with Unity 。
3.  My suggestion is : creat a build in defult Folder named "Current" to collect those project which taken from registry.
4. The custon Folder （perhaps named “GitHub”）those created by user，they canselect a specific path or drag a folder from Explorer into the “GitHub” folder then gather all the project and fill the Scrollview and then serialize all of the content into a json file mybe （any serialization solution is ok）.

sorry for my poor english,yours

**Comment by unitycoder** [comment]
now that we have custom history implemented (over 40 projects), can think about this favourites list again.

still unsure how to make the UI intuitive (and keep it compact)

![image](https://user-images.githubusercontent.com/5438317/206725584-3598fff9-9565-47dd-964a-5a676c86b350.png)

**Comment by Bian-Sh** [comment]
to be honest，I think the unity hub is good enough, and i like it, so my suggestion maybe not useful and have no constructive。

In my opinion, more less nformation is more less Sense of Pressure, you can use big icon, and less item of Listview.
And

star column is a good idea.

![image](https://user-images.githubusercontent.com/24520547/219376071-87c84f6e-c2bd-41b9-8624-6fb2010847a8.png)
and i think tabs that not used often should collect and gather together into a configuration button.
